### PR TITLE
[CELEBORN-1642][CIP-11] Support multiple worker tags

### PR DESCRIPTION
**Pull request by s0nskar**
### What changes were proposed in this pull request?
Current TagsManager code only supported one tags for selecting tagged workers. This change will enable support of passing multiple tags to TagsManager. Multiple tags will be evaluated as "AND" expression i.e only workers tagged with all the passed tags will be selected. 

Support for more schemes will be added in follow up PRs.

### Why are the changes needed?
https://cwiki.apache.org/confluence/display/CELEBORN/CIP-11+Supporting+Tags+in+Celeborn

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Added UTs

### CODE REVIEW

File: master/src/main/scala/org/apache/celeborn/service/deploy/master/tags/TagsManager.scala [hunk 1]
```diff
@@ -43,19 +43,27 @@ class TagsManager extends Logging {
...
+          workersForTags = Some(taggedWorkers)
+      }
+    }
+
+    if (workersForTags.isEmpty) {
+      logWarning(s"No workers for tags: $tagExpr found in cluster")
+      return Collections.emptyList()
     }
 
     val workerTagsPredicate = new Predicate[WorkerInfo] {
```
Comment (FMX) [inline master/src/main/scala/org/apache/celeborn/service/deploy/master/tags/TagsManager.scala:65]: This predicate can be merged into the filter method like the following codes.

`workers.stream().filter(w=>workersForTags.get.contains(w.toUniqueId())).collect(Collectors.toList())`

**Review by FMX — COMMENTED** [review]
LGTM overall. I think the code style can be changed a little.

File: master/src/main/scala/org/apache/celeborn/service/deploy/master/tags/TagsManager.scala (hunk 1 above)
Comment (s0nskar) [reply master/src/main/scala/org/apache/celeborn/service/deploy/master/tags/TagsManager.scala:65]: @FMX i tried this and couple of other things, but java8 was not happy. So had to go predicate route.

Comment (s0nskar) [reply master/src/main/scala/org/apache/celeborn/service/deploy/master/tags/TagsManager.scala]: Lost the old logs to trying it out once again here.

**Review by FMX — APPROVED** [review]
LGTM.

File: master/src/main/scala/org/apache/celeborn/service/deploy/master/tags/TagsManager.scala (hunk 1 above)
Comment (s0nskar) [reply master/src/main/scala/org/apache/celeborn/service/deploy/master/tags/TagsManager.scala]: This is the error –
```
[error] /home/runner/work/celeborn/celeborn/master/src/main/scala/org/apache/celeborn/service/deploy/master/tags/TagsManager.scala:65:29: missing parameter type
[error]     workers.stream().filter(w => workersForTags.get.contains(w.toUniqueId())).collect(
[error]                             ^
[error] one error found
```

Comment (FMX) [reply master/src/main/scala/org/apache/celeborn/service/deploy/master/tags/TagsManager.scala:65]: OK, I understand. Scala 2.11 doesn't support Java 8 lambda natively.

**Comment by FMX** [comment]
Merged into main(v0.6.0).

**Pull request closed without merging**